BedrockServer.cpp

```diff
@@ -991,11 +991,22 @@ void BedrockServer::worker(int threadId)
                         // Roll back the transaction, it'll get re-run in the sync thread.
                         core.rollback();
 
-                        // We're not handling a writable command anymore.
-                        SINFO("Sending non-parallel command " << command->request.methodLine
-                              << " to sync thread. Sync thread has " << _syncNodeQueuedCommands.size()
-                              << " queued commands.");
-                        _syncNodeQueuedCommands.push(move(command));
+                        if (state == SQLiteNode::LEADING) {
+                            SINFO("Sending non-parallel command " << command->request.methodLine
+                                  << " to sync thread. Sync thread has " << _syncNodeQueuedCommands.size() << " queued commands.");
+                            _syncNodeQueuedCommands.push(move(command));
+                        } else if (state == SQLiteNode::STANDINGDOWN) {
+                            SINFO("Need to process command " << command->request.methodLine << " but STANDINGDOWN, moving to _standDownQueue.");
+                            _standDownQueue.push(move(command));
+                        } else if (_clusterMessenger.sendToLeader(*command)) {
+                            SINFO("Escalating " << command->request.methodLine << " to leader.");
+                            waitForHTTPS(move(command));
+                        } else {
+                            // TODO: Something less naive that considers how these failures happen rather than a simple
```

Comment:
Won't this happen every time we have a leader switchover, i.e. twice in every deploy? Seems like we could do better in that predictable state than spamming the logs with these warns. Maybe there's a state variable of whether or not we have a leader that sendToLeader could update? Or maybe we just change this to a SHMM?

Reply:
It might. I haven't specifically tested this case. We shouldn't get this far if there is not leader, we need to be `FOLLOWING` to get to this code.

There may be a race condition where leader goes from `STANDINGDOWN` to disconnected and we still try and send a command to it. I hacven't verified this one way or the other. We could test for that case and handle it explicitly but I'm not sure if it even exists and this warning would catch it in production if it does.

Do you think it's worth spending the time on it before shipping this?

Reply:
ah, ok, that's a good point, when leader steps down, we'll probably go searching before very long.